translator/trace/zipkin/protospan_to_zipkinv1.go

```diff
@@ -295,29 +292,78 @@ func OCSpanDataToZipkin(
 	}
 
 	// construct Annotations from s.Annotations and s.MessageEvents.
-	if len(s.Annotations) != 0 || len(s.MessageEvents) != 0 {
-		z.Annotations = make([]zipkinmodel.Annotation, 0, len(s.Annotations)+len(s.MessageEvents))
-		for _, a := range s.Annotations {
-			z.Annotations = append(z.Annotations, zipkinmodel.Annotation{
-				Timestamp: a.Time,
-				Value:     a.Message,
-			})
+	z.Annotations = timeEventsToSpanAnnotations(s.TimeEvents)
+
+	return z, nil
+}
+
+func resourceToSpanAttributes(resource *resourcepb.Resource, tags map[string]string) {
+	if resource != nil {
+		for key, value := range resource.Labels {
+			tags[key] = value
+		}
+	}
+}
+
+func timeEventsToSpanAnnotations(timeEvents *tracepb.Span_TimeEvents) []zipkinmodel.Annotation {
+	var annotations []zipkinmodel.Annotation
+	// construct Annotations from s.Annotations and s.MessageEvents.
+	if timeEvents != nil && len(timeEvents.TimeEvent) != 0 {
+		annotations = make([]zipkinmodel.Annotation, 0, len(timeEvents.TimeEvent))
+		for _, te := range timeEvents.TimeEvent {
+			if te == nil || te.Value == nil {
+				continue
+			}
+
+			switch tme := te.Value.(type) {
+			case *tracepb.Span_TimeEvent_Annotation_:
+				annotations = append(annotations, zipkinmodel.Annotation{
+					Timestamp: internal.TimestampToTime(te.Time),
+					Value:     derefTruncatableString(tme.Annotation.Description),
+				})
+			case *tracepb.Span_TimeEvent_MessageEvent_:
+				a := zipkinmodel.Annotation{
+					Timestamp: internal.TimestampToTime(te.Time),
+					Value:     tme.MessageEvent.Type.String(),
```

Comment:
Does it matter that this will change from "RECV" to "RECEIVED"?

Reply:
Not that I can think of.